mlir/include/mlir/Dialect/SCF/IR/SCFOps.td

```diff
@@ -1121,7 +1121,7 @@ def IndexSwitchOp : SCF_Op<"index_switch", [RecursiveMemoryEffects,
     Example:
 
     ```mlir
-    %0 = scf.index_switch %arg0 : index -> i32
+    %0 = scf.index_switch %arg0 -> i32
```

Comment:
Seems like it's not following the usual convention, if we only print the result/operand we should use `:`, like below in: `arith.constant 10 : i32` and `scf.yield %1 : i32`. Can we fix this instead?

Reply:
> Seems like it's not following the usual convention, if we only print the result/operand we should use `:`, like below in: `arith.constant 10 : i32` and `scf.yield %1 : i32`. Can we fix this instead?

To fix that we need to change the parser's logic here: 
```
let assemblyFormat = [{
    $arg attr-dict (`->` type($results)^)?
    custom<SwitchCases>($cases, $caseRegions) `\n`
    `` `default` $defaultRegion
  }];
```
Moreover, many other already written examples use `->` instead of `:`.

Reply:
Hello, @joker-eph! What do you think about it? Should we change the assembly format in `.td`?

From:
```
  let assemblyFormat = [{
    $arg attr-dict (`->` type($results)^)?
    custom<SwitchCases>($cases, $caseRegions) `\n`
    `` `default` $defaultRegion
  }];
```
To: 
```
  let assemblyFormat = [{
    $arg attr-dict (`:` type($results)^)?
    custom<SwitchCases>($cases, $caseRegions) `\n`
    `` `default` $defaultRegion
  }];
```